src/cryptography/x509.py

```diff
@@ -1158,3 +1225,18 @@ def signature_hash_algorithm(self):
         Returns a HashAlgorithm corresponding to the type of the digest signed
         in the certificate.
         """
+
+
+@six.add_metaclass(abc.ABCMeta)
+class RevokedCertificate(object):
```

Comment:
Ah, right, reasons are technically part of an extensions sequence. My mistake. We should add extensions itself to the interface for certificate now though, that was an oversight. I'll put a PR in.